mesonbuild/coredata.py

```diff
@@ -783,15 +803,15 @@ def _set_others_from_buildtype(self, value: str) -> None:
 
     @staticmethod
     def is_per_machine_option(optname: OptionKey) -> bool:
-        if optname.name in BUILTIN_OPTIONS_PER_MACHINE:
+        if optname in BUILTIN_OPTIONS_PER_MACHINE:
```

Comment:
Okay, so this is buggy... but what did it do before? The function is used twice, once in mintro (I do not care what that does), and once in environment.py -- in the latter case, it is being used to "Keep only per machine options from the native file" Not sure I understand the ramifications of changing this.